price-estimator/src/filter.rs

```diff
@@ -40,13 +39,12 @@ pub fn all(
 struct NoTokenInfo;
 impl Reject for NoTokenInfo {}
 
-async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible> {
+async fn handle_rejection(err: Rejection) -> Result<impl Reply, Rejection> {
     let code;
     let message;
 
     if err.is_not_found() {
-        code = StatusCode::NOT_FOUND;
-        message = "invalid url path";
+        return Err(reject::not_found());
```

Comment:
The reason for this is that rejections should be used to signal that a filter cannot handle a request but another filter might. So we shouldn't return a response in the case of the path not matching so that the next filter can try to match the path.